pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -554,7 +564,12 @@ private void buildDedupMeta(ImmutableSegmentImpl immutableSegment) {
     PartitionDedupMetadataManager partitionDedupMetadataManager =
         _tableDedupMetadataManager.getOrCreatePartitionManager(partitionGroupId);
     immutableSegment.enableDedup(partitionDedupMetadataManager);
-    partitionDedupMetadataManager.addSegment(immutableSegment);
+    SegmentDataManager oldSegmentManager = _segmentDataManagerMap.get(segmentName);
```

Comment:
looks like you fixed `// TODO: Change dedup handling to handle segment replacement` commented above L539

Reply:
nit: perhaps rename buildDedupMeta to handleDedup() for a bit naming consistency with handleUpsert().

Reply:
👍